### PR TITLE
refactor payment request constructor tests

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
found a couple of buggy tests too.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258423255)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258423256)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258423262)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258423263)

# Sauce (MicrosoftEdge) #

File: payment-request/payment-request-constructor.https.html [hunk 1]
```diff
@@ -251,128 +208,93 @@
...
       new PaymentRequest(defaultMethods, details);
-    } catch (err) {
-      assert_true(false, `${prop} can be zero length`);
-    }
-  }, `PaymentDetailsBase.${prop} can be 0 length`);
-}
+      assert_unreached(`PaymentDetailsBase.${prop} can be zero length`);
+    } catch (err) {}
+  }
+}, `PaymentDetailsBase memebers can be 0 length`);
```
Comment (bentian) [inline payment-request/payment-request-constructor.https.html]: typo: mem' 'bers

File: payment-request/payment-request-constructor.https.html [hunk 2]
```diff
@@ -7,30 +7,51 @@
...
+// Avoid false positives, this should always pass
+function smokeTest() {
+  new PaymentRequest(defaultMethods, defaultDetails);
+}
 test(() => {
+  smokeTest();
   const request = new PaymentRequest(defaultMethods, defaultDetails);
-  assert_true(!!request.id, "must be some truthy value");
+  assert_true(Boolean(request.id), "must be some truthy value");
 }, "If details.id is missing, assign a identifier");
```
Comment (bentian) [inline payment-request/payment-request-constructor.https.html]: s/a/an

File: payment-request/payment-request-constructor.https.html [hunk 3]
```diff
@@ -52,168 +74,103 @@
...
+    {
+      supportedMethods: "¡basic-*-card!",
+    },
+  ];
+  for (const invalidMethod of invalidMethods) {
+    try {
+      new PaymentRequest([invalidMethod], defaultDetails);
+    } catch (err) {
+      assert_unreached(
+        `Unexpected error processing invalid PMI "${invalidMEthod}". ${err.message}`
```
Comment (bentian) [inline payment-request/payment-request-constructor.https.html]: invalidM'e'thod

File: payment-request/payment-request-constructor.https.html [hunk 4]
```diff
@@ -455,29 +368,30 @@
...
-        });
         new PaymentRequest(defaultMethods, details);
-      }
+      },
+      `Expected TypeError for option.amount.value: "${amount}"`
     );
-  }, "For each option in details.shippingOptions: if option.amount.value is not " + 'a valid decimal monetary value (in this case "' + amount + '"), then throw a TypeError');
-}
+  }
+}, `For each option in details.shippingOptions: if option.amount.value is not 'a valid decimal monetary value, then throw a TypeError`);
```
Comment (bentian) [inline payment-request/payment-request-constructor.https.html]: nit: extra ' before `a valid decimal ...`.

**Review by bentian — APPROVED** [review]

File: payment-request/payment-request-constructor.https.html [hunk 5]
```diff
@@ -30,6 +30,13 @@
   const request1 = new PaymentRequest(defaultMethods, defaultDetails);
   const request2 = new PaymentRequest(defaultMethods, defaultDetails);
   assert_not_equals(request1.id, request2.id, "UA generated ID must be unique");
```
Comment (marcoscaceres) [inline payment-request/payment-request-constructor.https.html]: It's a "fail fast". Technically, you are right that we probably don't need it ... but it's pretty cheap and gets the point across so I'm just going to leave it.

File: payment-request/payment-request-constructor.https.html (hunk 3 above)
Comment (marcoscaceres) [inline payment-request/payment-request-constructor.https.html]: Good catch!

**Comment by marcoscaceres** [comment]
@bentian, can you re-approve the review? That will allow me to merge (no, "carrying over r+ on GitHub" 😅)

**Review by bentian — APPROVED** [review]

**Comment by marcoscaceres** [comment]
my bad, was missing @bentian from OWNERS list https://github.com/w3c/web-platform-tests/pull/6671

**Comment by marcoscaceres** [comment]
@sideshowbarker, can you merge this please? I don't know why it's not recognizing Ben as an authorized person.

**Review by alphan102 — APPROVED** [review]

**Comment by marcoscaceres** [comment]
@jgraham or @sideshowbarker, can you grant the reviewers (and me) write access? Or are we violating the process by doing that. We'd like to be able to land these tests more quickly.

**Comment by sideshowbarker** [comment]
> @sideshowbarker, can you merge this please? I don't know why it's not recognizing Ben as an authorized person.

I think because @bentian didn’t have write access to the repo. But he does now (after accepting the invite)

**Comment by sideshowbarker** [comment]
> @jgraham or @sideshowbarker, can you grant the reviewers (and me) write access?

You already had write access and now @bentian and @edenchuang and @alphan102 do as well, so I believe you should be all set now.

**Comment by marcoscaceres** [comment]
Yay! thanks @sideshowbarker!!

**Pull request merged**